# Desafío: Estadística Descriptiva y Probabilidades (Parte I)
### Daniel Rojas López

### Ejercicio 1:

Se solicita calcular el promedio general de los salarios, desviación estándar, quintiles y rango utilizando la columna `salary_in_usd` para asegurar una comparación estandarizada.

In [5]:
import pandas as pd

# Cargar datos
df = pd.read_csv("ds_salaries.csv")

# Indicadores para salary_in_usd
mean_salary = df['salary_in_usd'].mean()
std_salary = df['salary_in_usd'].std()
quintiles = df['salary_in_usd'].quantile([0.2, 0.4, 0.6, 0.8])
salary_range = df['salary_in_usd'].max() - df['salary_in_usd'].min()

print(f"Promedio: ${mean_salary:,.2f} USD")
print(f"Desviación Estándar: ${std_salary:,.2f} USD")
print("Quintiles:")
print(quintiles)
print(f"Rango: ${salary_range:,.2f} USD")

Promedio: $137,570.39 USD
Desviación Estándar: $63,055.63 USD
Quintiles:
0.2     83972.8
0.4    120000.0
0.6    149850.0
0.8    185900.0
Name: salary_in_usd, dtype: float64
Rango: $444,868.00 USD


**Interpretación Ejercicio 1:**
- El **promedio** (~$137,570) nos da una idea del centro de la distribución, pero la **desviación estándar** (~$63,055) es bastante alta, lo que indica que los salarios varían significativamente respecto al promedio.
- El **rango** de más de $444k confirma la existencia de una gran brecha entre los salarios iniciales y los de niveles ejecutivos.
- Los **quintiles** nos permiten segmentar a la población: el 20% que menos gana percibe menos de $84k, mientras que el 20% superior supera los $185k.

### Ejercicio 2:

Se realiza una comparación entre los salarios agrupados por 3 categorías: `experience_level`, `company_size` y `employment_type`.

In [6]:
categories = ['experience_level', 'company_size', 'employment_type']

for cat in categories:
    print(f"\n--- Análisis por {cat} ---")
    stats = df.groupby(cat)['salary_in_usd'].agg(['mean', 'median', 'std', 'count'])
    # Calculamos Coeficiente de Variación (CV) para medir representatividad relativa
    stats['cv'] = stats['std'] / stats['mean']
    print(stats)


--- Análisis por experience_level ---
                           mean    median           std  count        cv
experience_level                                                        
EN                 78546.284375   70000.0  52225.424309    320  0.664900
EX                194930.929825  196000.0  70661.929661    114  0.362497
MI                104525.939130  100000.0  54387.685128    805  0.520327
SE                153051.071542  146000.0  56896.263954   2516  0.371747

--- Análisis por company_size ---
                       mean    median           std  count        cv
company_size                                                        
L             118300.982379  108500.0  75832.391505    454  0.641012
M             143130.548367  140000.0  58992.813382   3153  0.412161
S              78226.682432   62146.0  61955.141792    148  0.791995

--- Análisis por employment_type ---
                          mean    median            std  count        cv
employment_type                 

1. **¿Para cuál categoría las medidas son más representativas?**
Para el nivel de experiencia **'EX' (Executive)** y **'SE' (Senior)**. 
- Tienen los Coeficientes de Variación (CV) más bajos (0.36 y 0.37 respectivamente).
- Esto significa que los datos están más agrupados cerca del promedio y hay menos dispersión relativa.

2. **¿Para cuál son menos representativas?**
Para el tipo de contrato **'CT' (Contract)** y **'PT' (Part-time)**.
- Sus CV son muy elevados (> 0.9), lo que indica una dispersión extrema.
- Además, el tamaño de la muestra es muy pequeño (10 y 17 registros), lo que hace que cualquier valor atípico mueva mucho el promedio, restándole confianza.

### Ejercicio 3:

Considerando las medidas de tendencia central y dispersión calculadas:

1. **Tendencia Central:** La **mediana** suele ser una medida más robusta que la media en este dataset, especialmente en categorías con pocos datos como 'Small Companies', donde la media se ve inflada por salarios altos.
2. **Dispersión:** El dataset muestra una alta heterogeneidad. La desviación estándar global representa casi el 45% del valor del promedio, lo que sugiere que no se puede hablar de un "salario estándar" en Data Science sin especificar al menos el nivel de experiencia y el tamaño de la empresa.
3. **Conclusión:** A mayor nivel de experiencia y mayor tamaño de empresa (M o L), los indicadores tienden a ser más estables y representativos del mercado.